src/overlays/actors/ovl_En_Md/z_en_md.c

```diff
@@ -21,79 +57,813 @@ const ActorInit En_Md_InitVars = {
+                    gSaveContext.infTable[1] |= 0x200;
+                    break;
+                case 0x1033:
+                case 0x1067:
+                    return 2;
+            }
+            return 0;
+        case 5:
+            if (func_80106BC8(globalCtx) != 0) {
+                return 2;
+            }
+        default:
+            return 1;
+    }
+}
+
+u8 EnMd_ShouldSpawn(EnMd* this, GlobalContext* globalCtx) {
+    if (globalCtx->sceneNum == SCENE_SPOT04) {
+        if (!(gSaveContext.eventChkInf[1] & 0x1000) && !(gSaveContext.eventChkInf[4] & 1)) {
+            return 1;
+        }
+    }
+
+    if (globalCtx->sceneNum == SCENE_KOKIRI_HOME4) {
+        if (((gSaveContext.eventChkInf[1] & 0x1000) != 0) || ((gSaveContext.eventChkInf[4] & 1) != 0)) {
+            if (gSaveContext.linkAge != 0) {
+                return 1;
+            }
+        }
+    }
+
+    if (globalCtx->sceneNum == SCENE_SPOT10) {
+        return 1;
+    }
+
+    return 0;
+}
+
+void EnMd_UpdateEyes(EnMd* this) {
+    if (DECR(this->blinkCountdown) == 0) {
+        this->eyeImageIdx++;
+        if (this->eyeImageIdx > 2) {
+            this->blinkCountdown = Math_Rand_S16Offset(30, 30);
+            this->eyeImageIdx = 0;
+        }
+    }
+}
+
+void func_80AAB158(EnMd* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 absYawDiff;
+    s16 temp;
+    s16 temp2;
+    s16 yawDiff;
+
+    if (this->actor.xzDistFromLink < 170.0f) {
+        yawDiff = (f32)this->actor.yawTowardsLink - this->actor.shape.rot.y;
+        absYawDiff = ABS(yawDiff);
+
+        temp = (absYawDiff <= func_800347E8(2)) ? 2 : 1;
+        temp2 = 1;
+    } else {
+        temp = 1;
+        temp2 = 0;
+    }
+
+    if (this->unk_1E0.unk_00 != 0) {
+        temp = 4;
+    }
+
+    if (this->actionFunc == func_80AABD0C) {
+        temp = 1;
+        temp2 = 0;
+    }
+    if (this->actionFunc == func_80AAB8F8) {
+        temp = 4;
+        temp2 = 1;
+    }
+
+    if ((globalCtx->csCtx.state != 0) || (D_8011D394 != 0)) {
+        this->unk_1E0.unk_18 = globalCtx->view.eye;
+        this->unk_1E0.unk_14 = 40.0f;
+        temp = 2;
+    } else {
+        this->unk_1E0.unk_18 = player->actor.posRot.pos;
+        this->unk_1E0.unk_14 = (gSaveContext.linkAge > 0) ? 0.0f : -18.0f;
+    }
+
+    func_80034A14(&this->actor, &this->unk_1E0, 2, temp);
+    if (this->actionFunc != func_80AABC10) {
+        if (temp2) {
+            func_800343CC(globalCtx, &this->actor, &this->unk_1E0.unk_00, this->collider.dim.radius + 30.0f,
+                          func_80AAAE94, func_80AAAF04);
+        }
+    }
+}
+
+u8 func_80AAB370(EnMd* this, GlobalContext* globalCtx) {
+    Path* path;
+    s32 pad;
+    f32 xDiff;
+    f32 zDiff;
+    s32 temp_a2;
+    s32 temp_t7;
+    Vec3s* pointPos;
+
+    temp_a2 = this->actor.params & 0xFF00;
+    if (temp_a2 == 0xFF00) {
+        return 0;
+    }
+
+    temp_t7 = temp_a2 >> 8;
+    path = &globalCtx->setupPathList[temp_t7];
```

Comment:
You might be able to get rid of `temp_a2` and `temp_t7` by using `this->actor.params & 0xFF00` directly twice like this:
```suggestion
    if ((this->actor.params & 0xFF00) == 0xFF00) {
        return 0;
    }

    path = &globalCtx->setupPathList[(this->actor.params & 0xFF00) >> 8];
```
See En_Kz for reference.